src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -71,8 +78,18 @@ internal class CircuitHost : IAsyncDisposable
 
         public CircuitHandle Handle { get; }
 
+        /// <summary>
+        /// Gets a reference to the Circuit ID. This is an identifier that can be used safely in server-side
+        /// code to identify the circuit for logging, correlation or remote storage.
+        /// </summary>
         public string CircuitId { get; }
 
+        /// <summary>
+        /// Gets a reference to the Circuit Secret. This is an identifier that is used to make the handshake
+        /// between the client and server. This should not exposed to user-code.
+        /// </summary>
+        public CircuitSecret CircuitSecret { get; }
```

Comment:
I'm super open to feedback about the names used for these concepts. To me this was the clearest thing.

Note: the client-side code still uses `circuitId` as a name, but it always refers to circuit secret, not to circuitId. Open to discussion about renaming these where it's visible to the client as well.

Reply:
If it's not visible to the user, I wouldn't bother. CircuitSecret sounds fine to me, I would have thought that CircuitId and secret would have been paired together, why do we need to have them separate? I think of the CircuitSecret as the source of truth and the Id as a "DisplayName" for it, I would have tied them together.